spacy_llm/backends/dolly.py

```diff
@@ -79,8 +79,13 @@ def backend_dolly_hf(
         config = _compile_default_config()
 
     llm_pipeline = transformers.pipeline(model=model, **config)
+    return DollyBackend(llm_pipeline).query
```

Comment:
If we use `functools.partial` or a `lambda` function here instead, then the `_validate_types` function can't work. We need a class and a method here to track state and have valid type annotations.